tests/integration/kubernetes/lib.sh

```diff
@@ -78,7 +78,7 @@ assert_logs_contain() {
 
 	# Note: with image-rs we get more than the default 1000 lines of logs
 	print_node_journal "$node" "$log_id" --since "$datetime" -n 100000 \
-		grep "$message"
+		| grep "$message"
```

Comment:
Ohhh what a nasty bug ! Good catch Steve !

Does it fix something introduced by this PR ? If yes, I suggest you make this the first commit, for the sake of bisect again. :smiley:

Reply:
It took a long while to spot. The complexity is that without this change, doesn't actually break any tests. We use that function in a few different tests, including this, but the behaviour is that the grep fails, but test still passes, which is worrying, but explains why we missed this before.

I don't think it *needs* to go first as the tests pass without it (they just don't actually test the error message). I can move it up though if you would like.

Reply:
Done

Reply:
Hi @stevenhorsman @gkurz !

hmm... one test that uses `assert_logs_contain` is `k8s-measured-rootfs.bats` that run only if qemu-tdx. Guess what? With the fix now that test fails...

https://github.com/kata-containers/kata-containers/actions/runs/9893453384/job/27330646200?pr=9382
```
2024-07-11T15:53:36.7466161Z not ok 1 Test cannnot launch pod with measured boot enabled and incorrect hash
2024-07-11T15:53:36.7504034Z # (from function `assert_logs_contain' in file lib.sh, line 81,
2024-07-11T15:53:36.7505839Z #  in test file k8s-measured-rootfs.bats, line 55)
2024-07-11T15:53:36.7536942Z #   `assert_logs_contain "$node" kata "$node_start_time" \' failed
```

The `assert_logs_contain` is also used in k8s-guest-pull-image.bats which is only executed if qemu-coco-dev. However, the only test case that call the function (`Test we can pull a large image inside the guest`) is actually skipped in qemu-coco-dev... that's why we never noticed the problem.

` ok 2 Test we can pull a large image inside the guest # skip Test not supported for qemu-coco-dev`

Reply:
Ok, I'll add a commit to skip that!

Reply:
Hmm, so do we understand if measured root is just broken? I don't really want to scope creep this PR into trying to debug and fix that, so we could skip it and create an issue to track and enable it on all confidential systems?

Reply:
I've pushed a commit to print the logs for this test, in case it's just a log format issue and simple to fix

Reply:
Hi @stevenhorsman , I was looking at this problem and the best we can do is to follow your idea: disable the measured rootfs test; later  we decide what to do.

Longer discussion... the only kernel built for measured rootfs was the kernel-tdx-experimental, so that only the qemu-tdx job runs the test. The test check it cannot boot the VM and checks for a given message on the logs (using the `assert_logs_contain`). On commit 6cbdba72687683c2820d941905d934b8855baf62 we switched all TEE configurations to use the same kernel-confidential, so rootfs measured is disabled for qemu-tdx too now. The VM still fail to boot (because of a different reason...) but the bug on the `assert_logs_contain` is masking the checks on the logs.

Reply:
Ok, thanks for clarifying the kernel situation, so we don't actually expect this to pass now on any TEE, so skipping makes even more sense